### ISSUE TITLE
[graphiql] file upload support

### ISSUE THREAD

**Issue by Hebilicious** [graphql/graphiql#3511]
Support File Upload within GraphiQL out of the box.

GraphQL Altair supports something like this :

<img width="611" alt="image" src="https://github.com/graphql/graphiql/assets/13395944/315641a7-1e4a-4ae9-b9b4-849596eda071">

Uploaded files can be given a name and are then injected as variables in the query, where the File type can be used (assuming the schema as the scalar for it)

Support for multiple files as well  would be great.

**Comment by jaydenseric** [comment]
Here is the GraphQL multipart request specification:

https://github.com/jaydenseric/graphql-multipart-request-spec

A substantial number of GraphQL projects use this spec for file uploads in GraphQL operation variables, and for them GraphiQL is not a great option. Altair is going to make things a lot easier. It would be great to have more choice. The most recent iterations of the GraphiQL UI is quite nice and I would love to be able to use GraphiQL again after all these years if it can handle  uploading blobs via `Blob`, `File`, or `FileList` in the GraphQL operation variables.

**Comment by acao** [comment]
yes I have used it many times on both sides of the stack! this would be awesome to have. It has been requested before of course. The maintainer of Altair is a maintainer here as well! I think Apollo Studio or Workbench or whatever it's called also has support, I used it last year when we implemented your spec with nestjs.

i plan on making some improvements to the graphiql sdk later this year, but the goal would be to provide this via a plugin API, as many built in features will eventually become. in this case I think we also need to make the fetcher implementation pluggable as well. the biggest priority is still monaco of course. but I would love to do this right, and then make the other multipart spec we've implemented (IncrementalDelivery) a plugin as well
